addons/mail/static/src/component_hooks/use_messaging_features/use_messaging_features.js

```diff
@@ -0,0 +1,31 @@
+/** @odoo-module */
+
+import { useModels } from "@mail/component_hooks/use_models/use_models";
+import { useShouldUpdateBasedOnProps } from "@mail/component_hooks/use_should_update_based_on_props/use_should_update_based_on_props";
+
+const componentRegistry = {};
+
+/**
+ * Allows a component to lean on the messaging features: the component is
+ * subscribed to changes in messaging models in its environment, the component
+ * will only be rerenderd if its props change, and the component has access to
+ * all components that are registered with this function without having to
+ * explicitly import them.
+ *
+ * @param {function} ComponentClass the constructor of the component to be
+ *      registered. Its name will be used as its key in the registry.
+ * @param {Object} [propsCompareDepth={}] @see useShouldUpdateBasedOnProps
+ */
+export function registerMessagingComponent(ComponentClass, propsCompareDepth = {}) {
+    ComponentClass.components = Object.assign(Object.create(componentRegistry), ComponentClass.components);
+    const originalSetup = ComponentClass.prototype.setup;
+    ComponentClass.prototype.setup = function setup(...args) {
+        originalSetup.apply(this, args);
+        // useModels must be defined after useRenderedValues, indeed records and
+        // fields accessed during useRenderedValues should be observed by
+        // useModels as if they were part of the OWL rendering itself.
+        useModels();
+        useShouldUpdateBasedOnProps({ propsCompareDepth });
+    }
+    componentRegistry[ComponentClass.name] = ComponentClass;
```

Comment:
I'm not 100% sure if I like the name to be magic or not, but it should be fine for now. It does make for a lot less repetition currently, I guess we'll reconsider this if/when we don't need to specify the class manually.